### PR TITLE
Adding Functions e2e test template

### PR DESCRIPTION
**Pull request by aribeironovaes**
This contains Functions e2e test template + GetDevice Async after creating the device to make sure identity got created and cache updated.

### CODE REVIEW

File: smoke/IotEdgeQuickstart/details/Details.cs [hunk 1]
```diff
@@ -78,6 +78,15 @@ protected async Task GetOrCreateEdgeDeviceIdentity()
             {
                 await this.CreateEdgeDeviceIdentity(rm);
             }
+
+            //Adding a GetDeviceAsync again so it flushes the cash.
+            Device checkingDevice = await rm.GetDeviceAsync(this.deviceId);
```
Comment (varunpuranik) [inline smoke/IotEdgeQuickstart/details/Details.cs]: Did we confirm that this is needed and that it helps us?

File: smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_functions_deployment.template.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            "version": "1.0",
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "settings": {
+              "image": "edgebuilds.azurecr.io/microsoft/azureiotedge-functions-filter:<Build.BuildNumber>-linux-amd64",
+              "createOptions": ""
+            }
+          },
+
```
Comment (varunpuranik) [inline smoke/IotEdgeQuickstart/e2e_deployment_files/module_to_functions_deployment.template.json]: super nit - white space

File: smoke/IotEdgeQuickstart/details/Details.cs (hunk 1 above)
Comment (aribeironovaes) [reply smoke/IotEdgeQuickstart/details/Details.cs]: I don't think there is a way to confirm that. After I added the tests started passing. I can remove in the future. I also created a task under the e2e pbi so I can add to quickstart the capability to turn debug log on. That will help us in the future to debug.

Comment (aribeironovaes) [reply smoke/IotEdgeQuickstart/details/Details.cs]: Actually I will step back and remove that code. 

If tests keep failing I will investigate. Better than having a code that we don't need.

**Pull request merged**